### PR TITLE
Glue: Correlations minor APIs behavior improvements

### LINKED ISSUE #55377: Glue: Return correlation config in GET `/datasources/uid/{sourceUID}/correlations/{correlationUID}`
Returned data must include the configuration object.

### PR DESCRIPTION
**Pull request by Elfo404**
- `config.type` now always gets saved to the DB & returned by the API
- `config.type` defaults to `query` when not specified, or when config object is not specified
- when creating correlations, `config.type` is validated. as of now its only possible value is `query`
- when creating correlations, `config` is validated. if type==query and targetUID is not set it fails
- `CorrelationConfig` now sets a default target when being marshalled (empty object)
- add tests

Fixes https://github.com/grafana/grafana/issues/55377

### CODE REVIEW

File: pkg/services/correlations/models.go [hunk 1]
```diff
@@ -11,20 +13,49 @@ var (
...
+	return nil
+}
 
 // swagger:model
 type CorrelationConfig struct {
 	// Field used to attach the correlation link
 	// required:true
-	Field string `json:"field"`
+	Field string `json:"field" binding:"Required"`
 	// Target data query
```
Comment (ifrost) [inline pkg/services/correlations/models.go:37]: booo

Comment (ifrost) [inline pkg/services/correlations/models.go:37]:
```suggestion
	// Target type
```

**Review by ifrost — APPROVED** [review]
Good stuff, works as expected 👍  Just a small copy-pasted doc should be fixed.

Tested manually:
- [x] provisioning creates defined config 
- [x] provisioning creates an empty target in db if not defined
- [x] GET correlations API returns configs with specified targets
- [x] GET correlations API defaults to query type if type was not defined in db (old record)
- [x] POST correlations defaults target to an empty object if not provided
- [x] POST correlations validates when type=query and target are specified

**Comment by ifrost** [comment]
For the record - these changes are added as a follow up to https://github.com/grafana/grafana/issues/55378

**Comment by Elfo404** [comment]
@ifrost good call, updated (incidentally we also fixed https://github.com/grafana/grafana/issues/55377 as it returns correct data).

as of the `target` property name, i tried to come up with something less confusing but i can only think of `destination`. opinions?

**Comment by ifrost** [comment]
> as of the target property name, i tried to come up with something less confusing but i can only think of destination. opinions?

Now I'm not sure. I'm trying to describe the config in words and then find the word matching definitions for example:

_CorrelationConfig contains information used to build a data link. A data link uses CorrelationConfig to define custom behavior based on the CorrelationConfig (e.g. open a split view and run the query in the config)._

The word _destination_ suggests that what's defined there defines where the data link finishes, which may not be true because _how_ the _destination_ is used by the data link is defined in the front end.

Something that fits a bit more to me is for example `input` because it's used as an input to create the data link in front-end but I'm not fully convinced by any of them.

Maybe we can leave it as `target`? During the review I was confused by having `targetUid` and `target` but when I think about it now the confusion probably is more related to the former not the latter. If it was `targetDSUid`, I could say `targetDSUid` is part of the `target` data (just a property pulled out to a separate column). It's just that name `targetUid` doesn't make it clear that it's related to DS to me but I see why it was done this way ('sourceDataSource` looks also confusing too), so now I'm leaning more towards just leaving it as it is :D

**Pull request merged**